apps/omg/test/omg/state/core_test.exs

```diff
@@ -911,6 +913,227 @@ defmodule OMG.State.CoreTest do
            )
   end
 
+  describe "Automatic fees claiming" do
+    setup do
+      fee_claimer = OMG.TestHelper.generate_entity()
+
+      {:ok, child_block_interval} = OMG.Eth.RootChain.get_child_block_interval()
+      {:ok, state_empty} = Core.extract_initial_state(0, child_block_interval, fee_claimer.addr)
+
+      alice = OMG.TestHelper.generate_entity()
+      fees = %{@eth => %{amount: 2}}
+
+      state =
+        state_empty
+        |> do_deposit(alice, %{amount: 10, currency: @eth, blknum: 1})
+        |> Core.exec(create_recovered([{1, 0, 0, alice}], @eth, [{alice, 8}]), fees)
+        |> success?()
+
+      {:ok, [state: state, alice: alice, fees: fees, fee_claimer: fee_claimer, state_empty: state_empty]}
+    end
+
+    test "should append fee txs in block", %{state: state} do
+      state = Core.claim_fees(state)
```

Comment:
where does `form_block_check` come from? 
maybe we could start describing test steps and the intentions. For example:

The test starts with claiming and clearing any current fees.
_Not sure about form_block_check_
It continues to analyze all transactions in a block and asserts that it found a payment transaction and a fee transaction.

Reply:
`form_block_check` is a test helper (private fn declared at bottom) heavily used here.
I cleared the test intention - should be clear now.